arangod/Graph/Enumerators/WeightedShortestPathEnumerator.cpp

```diff
@@ -0,0 +1,884 @@
+      } else {
+        bestPath = std::make_tuple(nextFullPathWeight, otherStep, ourStep);
+      }
+    }
+  }
+}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+auto WeightedShortestPathEnumerator<
+    QueueType, PathStoreType, ProviderType,
+    PathValidator>::Ball::buildPath(Step const& vertexInShell,
+                                    PathResult<ProviderType, Step>& path)
+    -> void {
+  if (_direction == FORWARD) {
+    _interior.buildPath(vertexInShell, path);
+  } else {
+    _interior.reverseBuildPath(vertexInShell, path);
+  }
+}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+auto WeightedShortestPathEnumerator<QueueType, PathStoreType, ProviderType,
+                                    PathValidator>::Ball::provider()
+    -> ProviderType& {
+  return _provider;
+}
+
+/*
+ * Class: ResultCache (internally used in WeightedShortestPathEnumerator)
+ */
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+WeightedShortestPathEnumerator<
+    QueueType, PathStoreType, ProviderType,
+    PathValidator>::ResultCache::ResultCache(Ball& left, Ball& right)
+    : _internalLeft(left), _internalRight(right), _internalResultsCache{} {}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+WeightedShortestPathEnumerator<QueueType, PathStoreType, ProviderType,
+                               PathValidator>::ResultCache::~ResultCache() =
+    default;
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+auto WeightedShortestPathEnumerator<QueueType, PathStoreType, ProviderType,
+                                    PathValidator>::ResultCache::clear()
+    -> void {
+  _internalResultsCache.clear();
+};
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+auto WeightedShortestPathEnumerator<QueueType, PathStoreType, ProviderType,
+                                    PathValidator>::ResultCache::
+    tryAddResult(CalculatedCandidate const& candidate) -> bool {
+  auto const& [weight, first, second] = candidate;
+  PathResult<ProviderType, typename ProviderType::Step> resultPathCandidate{
+      _internalLeft.provider(), _internalRight.provider()};
+
+  // generates left and right parts of the path and combines them
+  // then checks whether we do have a path duplicate or not.
+  _internalLeft.buildPath(first, resultPathCandidate);
+  _internalRight.buildPath(second, resultPathCandidate);
+  for (auto const& pathToCheck : _internalResultsCache) {
+    bool foundDuplicate =
+        resultPathCandidate.isEqualEdgeRepresentation(pathToCheck);
+    if (foundDuplicate) {
+      return false;
+    }
+  }
+  _internalResultsCache.push_back(std::move(resultPathCandidate));
+  return true;
+};
```

Comment:
Function is unused

Reply:
Indeed the whole ResultCache class is no longer used. Removing it...